### PR TITLE
[NetAppFiles]Update to 2022-11-01

### PR DESCRIPTION
**Pull request by audunn**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

**Comment by audunn** [comment]
Not sure why the failure is happening in the pipeline, the failing test is not part of this PR `Azure.Core.Tests.ExponentialPolicyTest(False).DoesNotAdvanceHostWhenAdvancedByOtherThread`

**Comment by christothes** [comment]
You should rebase this on main to get the core and /eng changes.

**Comment by audunn** [comment]
Looks like the test-proxy push thing is not working for me.  Not sure how to fix this

**Comment by christothes** [comment]
> Looks like the test-proxy push thing is not working for me. Not sure how to fix this

//cc @kinelski

**Comment by JoshLove-msft** [comment]
> > Looks like the test-proxy push thing is not working for me. Not sure how to fix this
> 
> //cc @kinelski

/cc @scbedd

**Comment by pallavit** [comment]
/cc: @ArthurMa1978 as FYI since this is happening in control plane.

**Comment by pallavit** [comment]
@audunn you are likely hitting into permission issues, could you please tell us which microsoft team\org are you part of so we can find the right permission group to map you.

**Comment by scbedd** [comment]
> > > Looks like the test-proxy push thing is not working for me. Not sure how to fix this
> > 
> > 
> > //cc @kinelski
> 
> /cc @scbedd

This is just a permissions issue. @pallavit is working with @kurtzeborn to ensure the correct `azure-sdk-write` team gets created and @audunn will simply join that team when we have it created.

**Comment by audunn** [comment]
@pallavit I am with at NetApp on the Azure NetApp Files team.  
Thanks everyone for looking into this for me.

**Comment by audunn** [comment]
@pallavit 
I think I should already be in NetApp groups but we will double check. To be clear is this a MyAccess gorup that your are referring to?

**Comment by audunn** [comment]
@pallavit  can you clarify what more is needed.  For some reason I don't see that group in MyAccess.  I am currently a member of `Azure SDK Release Partners` , `TM-AzureNetApp`  and `Azure Standard Access`  Permission Level: RW

**Comment by pallavit** [comment]
This is the Azure teams group - https://github.com/orgs/Azure/teams/azure-netapp-files you need to request access.

**Comment by audunn** [comment]
FYI  azure-sdk-write-netapp did not exist it has now been created, https://github.com/orgs/Azure/teams/azure-sdk-write-netappfiles

**Comment by pallavit** [comment]
@audunn Thank you for the patience.

Now that we are past the asset move , /cc: @ArthurMa1978 - for help with review of the mgmt plane SDK.

**Comment by audunn** [comment]
@pallavit Do you mind elaborating a bit on what fixed it. Was it the azure-sdk-write-netapp group or something else?

**Comment by audunn** [comment]
@ArcturusZhang   Can you help identify the build issue cannot see where the problem is?

**Comment by scbedd** [comment]
> @pallavit Do you mind elaborating a bit on what fixed it. Was it the azure-sdk-write-netapp group or something else?

Exactly @audunn , our assets repo requires that you be part of one of the `write` groups. You hadn't been part of one up till that point, so we needed to create the appropriate group and assign you to it. As members of your team need to push additional assets, have them join that `netapp` group. Sorry the delay!

**Comment by audunn** [comment]
Thank you @scbedd

**Comment by audunn** [comment]
> @ArcturusZhang Can you help identify the build issue cannot see where the problem is?

hmm seems to have resolved itself.

**Comment by audunn** [comment]
@ArcturusZhang Can you take a look at this, can we move on with this now or is there more to be done?

File: sdk/netapp/Azure.ResourceManager.NetApp/api/Azure.ResourceManager.NetApp.netstandard2.0.cs [hunk 1]
```diff
@@ -359,6 +360,7 @@ public partial class NetAppVolumeData : Azure.ResourceManager.Models.TrackedReso
         public System.Collections.Generic.IReadOnlyList<Azure.ResourceManager.NetApp.Models.NetAppVolumeMountTarget> MountTargets { get { throw null; } }
         public Azure.ResourceManager.NetApp.Models.NetAppNetworkFeature? NetworkFeatures { get { throw null; } set { } }
         public System.Guid? NetworkSiblingSetId { get { throw null; } }
+        public string OriginatingResourceId { get { throw null; } }
```
Comment (ArcturusZhang) [inline sdk/netapp/Azure.ResourceManager.NetApp/api/Azure.ResourceManager.NetApp.netstandard2.0.cs:363]: Looks like this is a resource identifier, could we add the arm-id format to it?

File: sdk/netapp/Azure.ResourceManager.NetApp/api/Azure.ResourceManager.NetApp.netstandard2.0.cs [hunk 2]
```diff
@@ -709,6 +714,16 @@ public partial class CapacityPoolPatch : Azure.ResourceManager.Models.TrackedRes
         public static bool operator !=(Azure.ResourceManager.NetApp.Models.EnableNetAppSubvolume left, Azure.ResourceManager.NetApp.Models.EnableNetAppSubvolume right) { throw null; }
         public override string ToString() { throw null; }
     }
+    public partial class GetGroupIdListForLdapUserContent
+    {
+        public GetGroupIdListForLdapUserContent(string username) { }
+        public string Username { get { throw null; } }
+    }
+    public partial class GetGroupIdListForLdapUserResponse
```
Comment (ArcturusZhang) [inline sdk/netapp/Azure.ResourceManager.NetApp/api/Azure.ResourceManager.NetApp.netstandard2.0.cs:722]: shall we rename this to `GetGroupIdListForLdapUserResult`? Usually we should avoid `Response` suffix

File: sdk/netapp/Azure.ResourceManager.NetApp/api/Azure.ResourceManager.NetApp.netstandard2.0.cs [hunk 3]
```diff
@@ -1362,6 +1379,7 @@ public partial class NetAppVolumeGroupVolume
         public string Name { get { throw null; } set { } }
         public Azure.ResourceManager.NetApp.Models.NetAppNetworkFeature? NetworkFeatures { get { throw null; } set { } }
         public System.Guid? NetworkSiblingSetId { get { throw null; } }
+        public string OriginatingResourceId { get { throw null; } }
```
Comment (ArcturusZhang) [inline sdk/netapp/Azure.ResourceManager.NetApp/api/Azure.ResourceManager.NetApp.netstandard2.0.cs:1382]: Same here - this reads like a resource identifier from its name. Please change its format to arm-id

File: sdk/netapp/Azure.ResourceManager.NetApp/api/Azure.ResourceManager.NetApp.netstandard2.0.cs [hunk 4]
```diff
@@ -1403,6 +1421,7 @@ public partial class NetAppVolumePatch : Azure.ResourceManager.Models.TrackedRes
         public bool? IsCoolAccessEnabled { get { throw null; } set { } }
         public bool? IsDefaultQuotaEnabled { get { throw null; } set { } }
         public Azure.ResourceManager.NetApp.Models.NetAppFileServiceLevel? ServiceLevel { get { throw null; } set { } }
+        public bool? SnapshotDirectoryVisible { get { throw null; } set { } }
```
Comment (ArcturusZhang) [inline sdk/netapp/Azure.ResourceManager.NetApp/api/Azure.ResourceManager.NetApp.netstandard2.0.cs:1424]: Shall we rename this to `IsSnapshotDirectoryVisible`?

**Review by ArcturusZhang — COMMENTED** [review]
@audunn sorry for the delay of review - in general this PR looks good to me, I just left some minor comments about names of the new APIs, please take a look

**Comment by audunn** [comment]
@ArcturusZhang thanks for the review, can you have another look ?

**Review by ArcturusZhang — APPROVED** [review]

**Pull request merged**